Placement tenancy

csm opened the pull request:
Add placement tenancy option; UI for launching EBS-optimized instances. Still a WIP, pending testing.
- app/views/deployment/new.html: Add tenency selection.
- grails-app/conf/BuildConfig.groovy: update aws-java-sdk to 1.9.34.
- grails-app/controllers/com/netflix/asgard/AutoScalingController.groovy (save): set placement tenancy.
- grails-app/controllers/com/netflix/asgard/ClusterController.groovy (createNextGroup): set placement tenancy.
- grails-app/views/launchConfiguration/_launchConfigOptions.gsp: add UI elements for `ebsOptimized` and `placementTenancy`.
- src/groovy/com/netflix/asgard/model/LaunchConfigurationBeanOptions.groovy (placementTenancy): new field.
  (from, from, getCreateLaunchConfigurationRequest): pass along `placementTenancy`.
- src/groovy/com/netflix/asgard/push/GroupCreateOperation.groovy (start): set placement tenancy.
- src/groovy/com/netflix/asgard/push/GroupCreateOptions.groovy (placementTenancy): new field.
- src/groovy/com/netflix/asgard/push/RollingPushOperation.groovy (prepareGroup): pass in `placementTenancy`.
- test/unit/com/netflix/asgard/AutoScalingControllerSpec.groovy: set placement tenancy in expected results.
  Add test for setting placement tenancy.
- test/unit/com/netflix/asgard/DeploymentControllerSpec.groovy: set `placementTenancy` value.
- test/unit/com/netflix/asgard/EmailerServiceUnitSpec.groovy: fix matching of exception messages (this changed in the new SDK version).
- test/unit/com/netflix/asgard/HostedZoneControllerSpec.groovy: likewise.
- test/unit/com/netflix/asgard/deployment/StartDeploymentRequestSpec.groovy: add matching for `placementTenancy`.
- test/unit/com/netflix/asgard/model/LaunchConfigurationBeanOptionsSpec.groovy: add `placementTenancy`.

csmlyve committed:
Fix #602.

* grails-app/controllers/com/netflix/asgard/ImageController.groovy (ImageDeleteCommand): remove `configService`. Call into grailsApplication.config directly, it appears that configService doesn't get autowired in this command.

csmlyve committed:
Add placement tenancy option; UI for launching EBS-optimized instances.

* app/views/deployment/new.html: Add tenency selection.
* grails-app/conf/BuildConfig.groovy: update aws-java-sdk to 1.9.34.
* grails-app/controllers/com/netflix/asgard/AutoScalingController.groovy (save): set placement tenancy.
* grails-app/controllers/com/netflix/asgard/ClusterController.groovy (createNextGroup): set placement tenancy.
* grails-app/views/launchConfiguration/_launchConfigOptions.gsp: add UI elements for `ebsOptimized` and `placementTenancy`.
* src/groovy/com/netflix/asgard/model/LaunchConfigurationBeanOptions.groovy (placementTenancy): new field.
	(from, from, getCreateLaunchConfigurationRequest): pass along `placementTenancy`.
* src/groovy/com/netflix/asgard/push/GroupCreateOperation.groovy (start): set placement tenancy.
* src/groovy/com/netflix/asgard/push/GroupCreateOptions.groovy (placementTenancy): new field.
* src/groovy/com/netflix/asgard/push/RollingPushOperation.groovy (prepareGroup): pass in `placementTenancy`.
* test/unit/com/netflix/asgard/AutoScalingControllerSpec.groovy: set placement tenancy in expected results.
	Add test for setting placement tenancy.
* test/unit/com/netflix/asgard/DeploymentControllerSpec.groovy: set `placementTenancy` value.
* test/unit/com/netflix/asgard/EmailerServiceUnitSpec.groovy: fix matching of exception messages (this changed in the new SDK version).
* test/unit/com/netflix/asgard/HostedZoneControllerSpec.groovy: likewise.
* test/unit/com/netflix/asgard/deployment/StartDeploymentRequestSpec.groovy: add matching for `placementTenancy`.
* test/unit/com/netflix/asgard/model/LaunchConfigurationBeanOptionsSpec.groovy: add `placementTenancy`.

cloudbees-pull-request-builder commented:
[asgard-pull-requests #498](https://netflixoss.ci.cloudbees.com/job/asgard-pull-requests/498/) FAILURE
Looks like there's a problem with this pull request

csmlyve committed:
Fix code style issues.

csmlyve committed:
Fix.

cloudbees-pull-request-builder commented:
[asgard-pull-requests #499](https://netflixoss.ci.cloudbees.com/job/asgard-pull-requests/499/) FAILURE
Looks like there's a problem with this pull request

cloudbees-pull-request-builder commented:
[asgard-pull-requests #500](https://netflixoss.ci.cloudbees.com/job/asgard-pull-requests/500/) SUCCESS
This pull request looks good

csmlyve committed:
Fix issue with AsgardAWSCredentialsProviderChain.

* rc/groovy/com/netflix/asgard/cred/AsgardAWSCredentialsProviderChain.groovy (<init>): call `makeChain`.
	(makeChain): new method.

Groovy had trouble finding the right constructor for `STSAssumeRoleSessionCredentialsProvider`, and that constructor seems to disallow `null` arguments now.

csmlyve committed:
Set value to "true".

cloudbees-pull-request-builder commented:
[asgard-pull-requests #501](https://netflixoss.ci.cloudbees.com/job/asgard-pull-requests/501/) SUCCESS
This pull request looks good

cloudbees-pull-request-builder commented:
[asgard-pull-requests #502](https://netflixoss.ci.cloudbees.com/job/asgard-pull-requests/502/) SUCCESS
This pull request looks good
